bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/handler/MonopriceAudioHandler.java

```diff
@@ -0,0 +1,723 @@
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioSerialConnector;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioZone;
+import org.openhab.binding.monopriceaudio.internal.configuration.MonopriceAudioThingConfiguration;
+import org.openhab.binding.monopriceaudio.internal.dto.MonopriceAudioZoneDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MonopriceAudioHandler} is responsible for handling commands, which are sent to one of the channels.
+ *
+ * Based on the Rotel binding by Laurent Garnier
+ *
+ * @author Michael Lobstein - Initial contribution
+ */
+@NonNullByDefault
+public class MonopriceAudioHandler extends BaseThingHandler implements MonopriceAudioMessageEventListener {
+    private static final long RECON_POLLING_INTERVAL = TimeUnit.SECONDS.toSeconds(60);
+    private static final long INITIAL_POLLING_DELAY = TimeUnit.SECONDS.toSeconds(5);
+    private static final Pattern PATTERN = Pattern
+            .compile("^(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})");
+
+    private static final String ZONE = "ZONE";
+    private static final String ALL = "all";
+    private static final String CHANNEL_DELIMIT = "#";
+    private static final String ON = "01";
+    private static final String OFF = "00";
+
+    private static final int ONE = 1;
+    private static final int MAX_ZONES = 18;
+    private static final int MAX_SRC = 6;
+    private static final int MIN_VOLUME = 0;
+    private static final int MAX_VOLUME = 38;
+    private static final int MIN_TONE = -7;
+    private static final int MAX_TONE = 7;
+    private static final int MIN_BALANCE = -10;
+    private static final int MAX_BALANCE = 10;
+    private static final int BALANCE_OFFSET = 10;
+    private static final int TONE_OFFSET = 7;
+
+    private final Logger logger = LoggerFactory.getLogger(MonopriceAudioHandler.class);
+    private final MonopriceAudioStateDescriptionOptionProvider stateDescriptionProvider;
+
+    private @Nullable ScheduledFuture<?> reconnectJob;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private SerialPortManager serialPortManager;
+    private MonopriceAudioConnector connector = new MonopriceAudioDefaultConnector();
+
+    private ArrayList<String> ignoreZones = new ArrayList<>();
+    private long lastPollingUpdate = System.currentTimeMillis();
+    private long pollingInterval = 0;
+    private int numZones = 0;
+    private int allVolume = 1;
+    private int initialAllVolume = 0;
+    private Object sequenceLock = new Object();
+
+    // build a Map with a MonopriceAudioZoneDTO for each zoneId
+    private static final Map<String, MonopriceAudioZoneDTO> ZONE_DATA_MAP = MonopriceAudioZone.VALID_ZONE_IDS.stream()
+            .collect(Collectors.toMap(s -> s, s -> new MonopriceAudioZoneDTO()));
```

Comment:
Up to three can be connected together as one unit and accordingly up to 18 zones can be specified in the configuration. If they have more than 18 zones, they can always add another thing instance.

Reply:
If they add another thing instance then its handler would still be using this same static map. It should be a single map per handler right?

Reply:
> 
> 
> If they add another thing instance then its handler would still be using this same static map. It should be a single map per handler right?

Yeah, each instance of the handler should have its own map so I will remove static, but it can still be final.